experiment/simulation/js/display.js

```diff
@@ -0,0 +1,148 @@
+export let display = null;
+
+export function clearDisplay() {
+    display = null;
+}
+
+class Display {
+    constructor() {
+        this.id = "Display-" + window.numComponents++;
+        this.inputs = {
+            a: null,
+            b: null,
+            c: null,
+            d: null,
+            e: null,
+            f: null,
+            g: null
+        }
+        this.inputPoints = [];
+    }
+
+    generateComponent() {
+        const component = `<div id=${this.id} class="display"> <div class="display-container display-size-12 display-no"> <div id=${this.id}-segment-a class="segment-x segment-a segment-off"><span class="segment-border"></span></div> <div id= ${this.id}-segment-b class="segment-y segment-b segment-off"><span class="segment-border"></span></div> <div id=${this.id}-segment-c +' class="segment-y segment-c segment-off"><span class="segment-border"></span></div> <div id= ${this.id}-segment-d class="segment-x segment-d segment-off"><span class="segment-border"></span></div> <div id= ${this.id}-segment-e class="segment-y segment-e segment-off"><span class="segment-border"></span></div> <div id= ${this.id}-segment-f class="segment-y segment-f segment-off"><span class="segment-border"></span></div> <div id= ${this.id}-segment-g class="segment-x segment-g segment-off"><span class="segment-border"></span></div> </div></div>`;
+        return component;
+    }
+
+    registerComponent(workingArea, x = 0, y = 0) {
+        const width = document.getElementById(workingArea).offsetWidth;
+        const height = document.getElementById(workingArea).offsetHeight;
+        let scale = 900;
+        let yScale = 800;
+        x = (x / scale) * width;
+        y = (y / yScale) * height;
+
+        const el = document.getElementById(this.id);
+        el.style.left = x + "px";
+        el.style.top = y + "px";
+        registerGate(this.id, this);
+    }
+
+    setInput(input, value) {
+        this.inputs[input] = value;
+    }
+
+    addInputPoints(input) {
+        this.inputPoints.push(input);
+    }
+
+    display() {
+        let a = this.inputs.a;
+        let b = this.inputs.b;
+        let c = this.inputs.c;
+        let d = this.inputs.d;
+        let e = this.inputs.e;
+        let f = this.inputs.f;
+        let g = this.inputs.g;
+
+        const el = document.getElementById(this.id);
+        if(a == null || b==null || c==null || d==null || e==null || f==null || g==null) {
+            printErrors("Input points of the Display are not connected properly\n",el);
+            return;
+        }
+
+        if(a) {
+            document.getElementById(this.id + "-segment-a").classList.remove("segment-off");
+            document.getElementById(this.id + "-segment-a").classList.add("segment-on");
+        }
+        else {
+            document.getElementById(this.id + "-segment-a").classList.remove("segment-on");
+            document.getElementById(this.id + "-segment-a").classList.add("segment-off");
+        }

```

Comment:
Can we take lines 67-74 in a function with the element as parameter and call it for each element? I see a lot of code duplication here which can be easily avoided.
Also I saw classList.replace function somewhere, which may be a good fit here